app/src/main/java/uk/gov/onelogin/optin/domain/source/OptInRemoteSource.kt

```diff
@@ -0,0 +1,7 @@
+package uk.gov.onelogin.optin.domain.source
+
+import uk.gov.onelogin.optin.domain.model.AnalyticsOptInState
+
+interface OptInRemoteSource {
```

Comment:
It does. I'm not entirely sure I agree with the sonar lint reasoning for doing this to be honest. I deliberately made it not a SAM, because what if I need to add another method in the future? I then not only have to change each sub-class, but also all calling sites, that hardly seems "more maintainable" . 

The suggestion breaks YAGNI in my opinion. 
I will run sonar lint and apply fixes or suppressions with reasoning though.